capstone/src/main/java/com/google/sps/search/servlets/SearchResultsServlet.java

```diff
@@ -50,13 +70,55 @@ public void doGet(String userId, HttpServletRequest request, HttpServletResponse
       for (Entity result : pq.asIterable()) {
         try {
           users.add(User.fromEntity(result));
+          namesScore = this.runDijkstra(namesScore, userId, searchString, User.fromEntity(result));
```

Comment:
HIGH level, I'm concerned that every time somebody types in a new character, you're getting a list of names using your trie implementation AND following up that entire result list with a call to dijkstra for every user in the result list......

If you're going to go a pathfinding route, I would suggest precalculating and saving distances to the database, or at least caching results.

Reply:
This servlet is only called when the user presses enter to list the user results, so it'll only do the pathfinding once when the user presses enter and not at each new character. Caching would still be useful though